proofs/src/errors.rs

```diff
@@ -23,53 +41,68 @@ impl From<Box<bincode::ErrorKind>> for ProofError {
   fn from(err: Box<bincode::ErrorKind>) -> ProofError { ProofError::Bincode(*err) }
 }
 
+/// Represents the various error conditions that can occur within the `proofs` crate.
 #[derive(Debug, Error)]
 pub enum ProofError {
+  /// The error is a bellpepper_core::SynthesisError
   #[error(transparent)]
   Synthesis(#[from] bellpepper_core::SynthesisError),
 
+  /// The error is a std::io::Error
   #[error(transparent)]
   Io(#[from] std::io::Error),
 
+  /// The error is a serde_json::Error
   #[error(transparent)]
   Serde(#[from] serde_json::Error),
 
+  /// The error is a custom error with a message
   #[error("Other error: {0}")]
   Other(String),
 
+  /// The error is a failed proof verification
   #[error("Failed to verify proof")]
   VerifyFailed(),
 
+  /// The error is a num_bigint::ParseBigIntError
   #[error(transparent)]
   Parse(#[from] num_bigint::ParseBigIntError),
 
+  /// The error is a WitnessCalcError
   #[cfg(not(target_arch = "wasm32"))]
   #[error(transparent)]
   WitnessCalc(#[from] WitnessCalcError),
 
+  /// The error is a missing header section
   #[error("Missing header section")]
   MissingSection,
 
+  /// The error is a bincode::ErrorKind
   #[error(transparent)]
   Bincode(#[from] bincode::ErrorKind),
 
+  /// The error is a client_side_prover::supernova::error::SuperNovaError
   #[error(transparent)]
   SuperNova(#[from] client_side_prover::supernova::error::SuperNovaError),
 
+  /// The error is a json key error
   #[error("json key not found: {0}")]
   JsonKeyError(String),
 
+  /// The error is a web_proof_circuits_witness_generator::WitnessGeneratorError
   #[error(transparent)]
   WitnessGenerator(#[from] web_proof_circuits_witness_generator::WitnessGeneratorError),
 
+  /// The error is an invalid circuit size
   #[error("Invalid circuit size")]
   InvalidCircuitSize,
 
+  /// The error is a serde_wasm_bindgen::Error
   #[cfg(target_arch = "wasm32")]
-  #[error("transparent")]
```

Comment:
oof lol

Reply:
yeah found that randomly, not sure if was giving us pain but at this this fixes it